File: challenge-9/count_vowel_words.py
# Python3 program to count the number of words
# in input string that start with a vowel
# Input Constraint: max_string_size = 500

import sys

# Count num of words starting with vowel in param
def count_vowel_words(check_string):

	count = 0

	# Generate list of words, on basis of ' '
	list_of_words = check_string.split(' ')

	# check each word in the list
	for word in list_of_words:

		# For extra spaces
		if len(word) == 0:
			continue
			
		# 0th index contains the first letter for each
		# word. Converting to lowercase ensures that capitals
		# are counted too
		if word[0].lower() in ['a', 'e', 'i', 'o', 'u']:
			count += 1

	print(f'Count: {count}')


# __main__ code

string = input('Enter String: ')

# If input constraint violated, exit. Also exit on empty string
if len(string) > 500 or len(string) == 0:
	print('Error: Invalid lenght of string')
	sys.exit(1)

count_vowel_words(string)

File: challenge-9/CountVowelWords.java
/*
* Java program to count the number of words in an input
* string that start with a vowel
* Input constraint: maximum string size = 500
*/

import java.util.Scanner;

public class CountVowelWords {

	// Count the number of variabls 
	private void countVowelWords(String check_string) {

		int vowelCount = 0;

		// Generate a list of words, split on the basis of ' '
		String[] listOfWords = check_string.split(" ", 0);

		// Check for each word in the list
		for (String word : listOfWords) {

			// Skip if extra space
			if (word.length() == 0) {
				continue;
			}

			// If character at 0th index of word is vowel
			if (isVowel(word.charAt(0)))
			{
				vowelCount++;
			}
		}

		System.out.println("Count: " + vowelCount);
	}

	// Returns true if param is vowel
	boolean isVowel(char letter) {

		// Convert to lower case to check capitalised letters too
		switch (Character.toLowerCase(letter)) {
			case 'a':
			case 'e':
			case 'i':
			case 'o':
			case 'u': return true;
			default: return false;
		}
	}


	// Driver code
	public static void main(String[] args) {

		// Scanner object for input
		Scanner scanner = new Scanner(System.in);

		System.out.print("Input String: ");
		String string = scanner.nextLine();


		// Throw error if input constraint violated
		assert string.length() > 500: "Invalid size of input string";

		// Class object for accessing member
		CountVowelWords count = new CountVowelWords();

		count.countVowelWords(string);
	}
}